src/groundlight/cli.py

```diff
@@ -14,9 +16,19 @@ def class_func_to_cli(method):
     Given the class method, simplify the typing on the inputs so that Typer can accept the method
     """
 
-    @wraps(method)
+    # We create a temporary class to bind the method to so we have the correct annotations for typer to use
+    # The first thing we'll do when we actually call the method is to bind it to a proper Groundlight instance
```

Comment:
This is some of the wildest python I've ever written. My conclusion is that if we want to make any significant changes to this behavior in the future we immediately ditch Typer and write our own argparser. Sans that, this does seem to do what we want it to do.